### DISCUSSION TITLE
Subject: Seeking Advice on Thermal and Day Camera Dataset Collection for Object Detection Accuracy (Class-Specific)

### DISCUSSION THREAD

**Discussion by nileshmetakota-17** [ultralytics/ultralytics#22415] [Solutions]
Message:
Hello Ultralytics team,

I'm currently working on an object detection project using both thermal and daytime images, with the goal of detecting specific classes (e.g., pedestrians, vehicles, animals, etc.). To achieve the best accuracy across these different modalities, I’m seeking guidance on the following:

Best practices for collecting and annotating datasets with clear, class-specific labeling for both thermal and day camera images.

Recommendations on balancing the dataset for each class, especially when some classes might be harder to capture in thermal imagery (like small objects or low-contrast targets).

Advice on handling challenges like temperature variations, low-light conditions, or background clutter, particularly with respect to specific classes.

Data augmentation strategies that can help improve model performance across both camera types, and maintain accuracy for each class in diverse conditions.

Any pre-processing tips or considerations for class-specific accuracy in object detection using thermal and day images.

Your expertise and any additional suggestions for optimizing detection accuracy across class-specific targets would be greatly appreciated. Thank you in advance!

Best regards,
[Nilesh]

**Comment by UltralyticsAssistant** [comment]
👋 Hello @nileshmetakota-17, thanks for starting this discussion and for your interest in Ultralytics 🚀! This is an automated response to help you get started quickly—an Ultralytics engineer will also review your post and assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
- Dataset details (e.g., sample images from both thermal and day cameras, class list, class balance/counts)
- Annotation format and examples
- Training command, hyperparameters, and augmentation settings
- Full training logs and environment info
And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Providing these details will help us give targeted guidance for class-specific accuracy across modalities 💡.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```

## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Reply by glenn-jocher** [reply]
Thanks Nilesh—great project. For collection/annotation, keep a single class taxonomy across modalities, time‑sync captures, and if sensors are rigidly mounted, calibrate/align RGB↔thermal; the guidance in the Ultralytics article on [data collection and annotation strategies](https://docs.ultralytics.com/guides/data-collection-and-annotation/) covers class design, active learning, and minimum per‑class counts that scale well to this setup. To handle skewed classes (often an issue in thermal), audit per‑class‑per‑modality counts, then oversample scarce thermal classes and iterate with slice‑level evaluation to avoid dataset bias; see [dataset bias and mitigation](https://www.ultralytics.com/glossary/dataset-bias). For augmentation, Albumentations works well: use contrast/CLAHE, noise, and mild blur for thermal, plus standard spatial transforms; details and examples are in the guide on [YOLO data augmentation with Albumentations](https://www.ultralytics.com/blog/using-albumentations-augmentations-to-diversify-your-data). If you prefer one model for both sensors, pack aligned RGB+thermal into multi‑channel TIFF and set `channels` in your dataset YAML following the [COCO8‑Multispectral (multi‑channel) example](https://docs.ultralytics.com/datasets/detect/coco8-multispectral/); otherwise, train one YOLO11 per modality and fuse detections downstream. For thermal preprocessing, normalize to a fixed 8‑bit range per camera (avoid per‑image auto‑contrast at train/test mismatch), keep resolution/FOV consistent, and use mosaic sparingly to help small objects in limited datasets (as shown in the multispectral example). Minimal setups:

```yaml
# thermal.yaml
path: datasets/thermal
train: images/train
val: images/val
channels: 1
names:
  0: person
  1: vehicle
  2: animal
```

```bash
yolo detect train data=thermal.yaml model=yolo11n.pt imgsz=640 epochs=100
yolo detect train data=rgb.yaml     model=yolo11n.pt imgsz=640 epochs=100
```

If low‑light edge cases are a focus, this short read on [nighttime detection considerations](https://www.ultralytics.com/blog/traffic-video-detection-at-nighttime-a-look-at-why-accuracy-is-key) may help prioritize what to collect. Please keep follow‑ups in this thread (no DMs); if you can share a tiny sample from each modality plus your exact train command, we can suggest class‑specific tweaks.
